test/datasets_utils.py

```diff
@@ -868,9 +868,12 @@ def _split_files_or_dirs(root, *files_or_dirs):
 def _make_archive(root, name, *files_or_dirs, opener, adder, remove=True):
     archive = pathlib.Path(root) / name
     if not files_or_dirs:
-        dir = archive.with_suffix("")
-        if dir.exists() and dir.is_dir():
-            files_or_dirs = (dir,)
+        # We cannot use Path.with_suffix("") here, since that only applies to the last suffix if multiple suffixes are
+        # present. For example, `pathlib.Path("foo.tar.gz").with_suffix("")` results in `foo.tar`. Since we want to
+        # extract the name without any suffixes here, we replace them manually.
+        file_or_dir = archive.with_name(archive.name.replace("".join(archive.suffixes), ""))
```

Comment:
Alternative as discussed offline

```suggestion
        file_or_dir = archive
        for _ in range(len(path.suffixes)):
            file_or_dir = file_or_dir.with_suffix("")
```

I'm fine with either.